### PR TITLE
feat(www): add back typescript indicator

### PR DESCRIPTION
**Pull request by Haroenv**
**What's the problem this PR addresses?**

This is present in v1, but was added after the v2 site was forked.

**How did you fix it?**

Added the indicator back, with the same css as in v1

I had to apply some changes from #739, so let's be careful if it's still correct, and in which order things get merged

### CODE REVIEW

**Comment by arcanis** [comment]
I get a `Uncaught TypeError: Cannot read property 'ts' of undefined` when starting to type

File: packages/gatsby/src/components/search/SearchProvider.js [hunk 1]
```diff
@@ -32,14 +32,15 @@ export const SearchProvider = ({searchState, onSearchStateChange, children}) =>
...
-            'repository',
             'homepage',
             'humanDownloadsLast30Days',
             'keywords',
             'license',
             'modified',
             'name',
             'owner',
+            'repository',
+            'typescript',
```
Comment (Haroenv) [inline packages/gatsby/src/components/search/SearchProvider.js]:
```suggestion
            'types',
```

Comment (Haroenv) [reply packages/gatsby/src/components/search/SearchProvider.js]: shouldn't have retyped it here 😂

**Comment by Haroenv** [comment]
I miscopied something @arcanis

**Comment by arcanis** [comment]
Sounds good! Btw it's interesting, some packages don't seem to be detected as being covered by TS types. For example [`clipanion`](https://deploy-preview-742--yarn2.netlify.com/package/clipanion) has them, but they aren't shown. I think it might be because the index doesn't take the `main` field into account when `types` is missing 🤔

**Comment by Haroenv** [comment]
clipanion doesn't seem to be detected, since `index` isn't added to folders it the detection I think. Will have a look later

**Pull request merged**